packages/react-router/modules/Route.js

```diff
@@ -98,7 +104,7 @@ class Route extends React.Component {
 
   componentWillReceiveProps(nextProps) {
     Object.assign(this.router, {
-      match: computeMatch(this.router, nextProps)
+      match: computeMatch(this.context.router, nextProps)
```

Comment:
Need to use the parent `router` object, not `this.router` in order to have access to the parent `match`.